api/supabase/queries/plant_by_id.ts

```diff
@@ -0,0 +1,19 @@
+import { UUID } from 'crypto';
+import { Plant } from '@/types/schema';
+import supabase from '../createClient';
+
+export async function get_plant_by_id(
+  i_state: string,
+  p_id: UUID,
+): Promise<Plant> {
+  const { data, error } = await supabase.rpc('get_plant_by_id', {
+    input_state: i_state,
+    plant_id: p_id,
+  });
+
+  if (error) {
+    throw new Error(`Error getting matching plant: ${error.message}`);
+  }
+
+  return data;
+}
```

Comment:
it seems like it's more reliable to not use supabase.rpc and just write the query directly... 

i tried with the following code and it worked!
```
export async function getPlantById(
  input_state: string,
  plant_id: UUID, 
): Promise<Plant> {
  const { data, error } = await supabase
    .from('plant_seasonality')
    .select('*')
    .eq('state', input_state)
    .eq('id', plant_id);
  if (error)
    throw new Error(`Error fetching plant seasonality: ${error.message}`);
  return data[0];
}
```

Reply:
edit: please rebase, as i've refactored this function already!